src/formatDate.js

```diff
@@ -50,7 +50,59 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const firstDivider = fromFormat[3];
+  const secondDivider = toFormat[3];
+  const originalDate = date.split(firstDivider);
+
+  const reverseDate = [];
+  let year;
+  let month;
+  let day;
+
+  for (let i = 0; i < 3; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = originalDate[i];
+        break;
+
+      case 'MM':
+        month = originalDate[i];
+        break;
+
+      case 'YYYY':
+        year = originalDate[i];
+        break;
+
+      case 'YY':
+        if (+originalDate[i] < 30) {
+          year = 20 + originalDate[i];
+        } else {
+          year = 19 + originalDate[i];
+        }
+        break;
+    }
+  }
+
+  for (let i = 0; i < 3; i++) {
+    switch (toFormat[i]) {
+      case 'DD':
+        reverseDate.push(day);
+        break;
+      case 'MM':
+        reverseDate.push(month);
+        break;
+      case 'YYYY':
+        reverseDate.push(year);
+        break;
+      case 'YY':
+        reverseDate[i] = year.slice(2);
+        break;
+    }
```

Comment:
and here `toFormat[i]` will be the key of your object

Reply:
I have made the changes you suggested, but, anyway, what is the point of the whole thing? It is supposed to make my code more readable and understandable, but does it in this case? Could you explain that, please?

Reply:
Yes. You store all values in 1 object and then dynamically (based on key (YY, DD, etc)) can write and access the value. This code will be more optimised and fast